src/main/java/org/jabref/model/entry/AuthorListParser.java

```diff
@@ -333,85 +339,188 @@ private String concatTokens(List<Object> tokens, int start, int end, int offset,
      * <CODE>token_case</CODE>.
      */
     private int getToken() {
+        visited[0] = true;
         tokenStart = tokenEnd;
         while (tokenStart < original.length()) {
+            visited[1] = true;
             char c = original.charAt(tokenStart);
             if (!((c == '~') || (c == '-') || Character.isWhitespace(c))) {
+                visited[2] = true;
                 break;
             }
+            else {
+                visited[3] = true;
+            }
             tokenStart++;
         }
         tokenEnd = tokenStart;
         if (tokenStart >= original.length()) {
+            visited[4] = true;
             return TOKEN_EOF;
```

Comment:
You could move the file writing to a separate function and call it before each return statement in order to make sure that every branch is correctly represented in the coverage file.

Reply:
There are some return statements that need to call `branchCoverage()` beforehand!